lib/livebook/runtime/erl_dist/runtime_server.ex

```diff
@@ -329,4 +396,31 @@ defmodule Livebook.Runtime.ErlDist.RuntimeServer do
   defp report_memory_usage(state) do
     send(state.owner, {:runtime_memory_usage, Evaluator.memory()})
   end
+
+  defp report_small_cell_definitions(state) do
+    smart_cell_definitions = get_smart_cell_definitions()
+
+    if smart_cell_definitions == state.smart_cell_definitions do
+      state
+    else
+      defs = Enum.map(smart_cell_definitions, &Map.take(&1, [:kind, :name]))
+      send(state.owner, {:runtime_smart_cell_definitions, defs})
+      %{state | smart_cell_definitions: smart_cell_definitions}
+    end
+  end
+
+  def __smart_cell_start__({mod, fun, args}, gl) do
+    Process.group_leader(self(), gl)
```

Comment:
There is another trick I believe you can do. You can implement the `init` callback of the DynamicSupervisor and change the group leader of the DynamicSupervisor. Then all of its children will inherit the group leader.

Reply:
That was my initial thought, it's just that this way we have one module less. But now I'm thinking we could also set the supervisor group leader once we start it, so then we get both. Any preference?